storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java

```diff
@@ -446,6 +447,7 @@ public void sendSupervisorAssignments(SupervisorAssignments assignments) {
     public void close() {
         try {
             LOG.info("Shutting down supervisor {}", getId());
+            metricsReporters.close();
```

Comment:
Nit: This should probably have a null check, since metricsReporters isn't guaranteed to be non-null.